src/scripts/main.js

```diff
@@ -1 +1,81 @@
 'use strict';
+
+const body = document.querySelector('body');
+
+const firstPromise = new Promise((resolve, reject) => {
+  // eslint-disable-next-line no-shadow
+  const onLeftClick = (event) => {
+    if (event.button === 0) {
+      resolve('First promise was resolved');
+    }
+  };
+
+  document.addEventListener('mousedown', onLeftClick);
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+  });
+
+  // eslint-disable-next-line no-shadow
+  document.addEventListener('contextmenu', (event) => {
+    event.preventDefault();
+    resolve('Second promise was resolved');
+  });
+});
+
+const thirdPromise = new Promise((resolve, reject) => {
+  let isLeftClicked = false;
+  let isRightClicked = false;
+
+  const onBothClick = () => {
+    if (isLeftClicked && isRightClicked) {
+      resolve('Third promise was resolved');
```

Comment:
The third promise resolves when both isLeftClicked and isRightClicked are true. However, if the user clicks one button, then the other, and then the first again, the promise will resolve incorrectly. Consider resetting the flags after resolution or adding additional logic to handle this scenario.